### PR TITLE
Add BloodHound customqueries support to my-resources (merge/replace)

### PR DESCRIPTION
**Pull request by gbe**
Following the work done in https://github.com/ThePorgs/Exegol-images/pull/163, and the release of `bqm 1.5`, I figured it would be easier for @Dramelac and others to review from a clean branch, and rebased from the latest version of `dev`.

Following the comments from https://github.com/ThePorgs/Exegol-images/pull/163, this PR adds support:
- to `merge` Exegol's default customqueries.json file with valid Bloodhound local json files a user places in `my-resources`,
- to `replace` Exegol's default customqueries.json file with valid Bloodhound local json files a user places in `my-resources`.

### CODE REVIEW

**Comment by Dramelac** [comment]
Thank you for the PR, I will review it as soon as possible!

**Review by ShutdownRepo — APPROVED** [review]
Seems fine to me, waiting for @Dramelac review

File: sources/assets/exegol/skel/bloodhound_customqueries/README.txt
Comment (Dramelac) [inline sources/assets/exegol/skel/bloodhound_customqueries/README.txt:1]: I think the folder can just be named "bloodhound", no need to add complexity here imo

File: sources/assets/exegol/skel/bloodhound_customqueries/merge/README.txt
Comment (Dramelac) [inline sources/assets/exegol/skel/bloodhound_customqueries/merge/README.txt:1]: A more specific message maybe ? "in this directory" + the extension of the file to add ?

File: sources/assets/exegol/skel/bloodhound_customqueries/replacement/README.txt
Comment (Dramelac) [inline sources/assets/exegol/skel/bloodhound_customqueries/replacement/README.txt:1]: Same as the comment above

File: sources/assets/exegol/skel/bloodhound_customqueries/README.txt
Comment (Dramelac) [inline sources/assets/exegol/skel/bloodhound_customqueries/README.txt:4]: Can you use Markdown extension & syntax on your files ? To use the same as the others modules.

File: sources/assets/exegol/load_supported_setups.sh
Comment (Dramelac) [inline sources/assets/exegol/load_supported_setups.sh:180]: Just `deploy_bloodhound` ?

File: sources/assets/exegol/load_supported_setups.sh
Comment (Dramelac) [inline sources/assets/exegol/load_supported_setups.sh:139]: Can you use `~/` instead of `/root/` ?

File: sources/assets/exegol/load_supported_setups.sh
Comment (Dramelac) [inline sources/assets/exegol/load_supported_setups.sh:149]: Same here: 
Can you use `~/` instead of `/root/` ?

File: sources/assets/exegol/load_supported_setups.sh
Comment (Dramelac) [inline sources/assets/exegol/load_supported_setups.sh:156]: Is there a bloodhound config file ? If so, we can also offer the option to the user to overwrite the config file.

File: sources/assets/exegol/load_supported_setups.sh
Comment (Dramelac) [inline sources/assets/exegol/load_supported_setups.sh:137]: You must handle what happen if not: create directory, copy skel file etc.

File: sources/assets/exegol/load_supported_setups.sh
Comment (Dramelac) [inline sources/assets/exegol/load_supported_setups.sh:147]: You must handle what happen if not: create directory, copy skel file etc.

File: sources/assets/exegol/load_supported_setups.sh
Comment (Dramelac) [inline sources/assets/exegol/load_supported_setups.sh:157]: Before running functions to handle bqm, you could check if the `bloodhound` directory exist and create it / copy it from skel if not.

**Review by Dramelac — CHANGES_REQUESTED** [review]
Hello,

Thank you very much for your PR, this is a very nice feature!
Here is some comments but minor updates overall, i will merge it as soon as everything is ready :)

**Comment by Dramelac** [comment]
PS (maintainers info): no need for workflow on this PR, it will not alter the build of the image, but it should be tested manually to see if everything work as expect when using `my-resources`

File: sources/assets/exegol/skel/bloodhound_customqueries/README.txt
Comment (gbe) [reply sources/assets/exegol/skel/bloodhound_customqueries/README.txt]: I renamed the folder to `bloodhound` as suggested.
However, as you rightfully requested the support for the config.json file. I thought it was not clear without the distinction.
For this reason, the sub folder `merge` and `replace` were renamed `customqueries_merge`.

File: sources/assets/exegol/skel/bloodhound_customqueries/merge/README.txt
Comment (gbe) [reply sources/assets/exegol/skel/bloodhound_customqueries/merge/README.txt]: I renamed the file README.md, added the markdown and tried to rework the content. Hopefully it's clearer that way.

File: sources/assets/exegol/skel/bloodhound_customqueries/replacement/README.txt
Comment (gbe) [reply sources/assets/exegol/skel/bloodhound_customqueries/replacement/README.txt]: Same as well

File: sources/assets/exegol/skel/bloodhound_customqueries/README.txt
Comment (gbe) [reply sources/assets/exegol/skel/bloodhound_customqueries/README.txt]: As mentioned above, this is done.

File: sources/assets/exegol/load_supported_setups.sh
Comment (gbe) [reply sources/assets/exegol/load_supported_setups.sh]: Done.

File: sources/assets/exegol/load_supported_setups.sh
Comment (gbe) [reply sources/assets/exegol/load_supported_setups.sh]: Done. I changed another path in the same file, and unrelated to bloodhound.

File: sources/assets/exegol/load_supported_setups.sh
Comment (gbe) [reply sources/assets/exegol/load_supported_setups.sh]: Done.

File: sources/assets/exegol/load_supported_setups.sh
Comment (gbe) [reply sources/assets/exegol/load_supported_setups.sh:169]: Agreed, I added support for it as well in the latest commit.

File: sources/assets/exegol/load_supported_setups.sh
Comment (gbe) [reply sources/assets/exegol/load_supported_setups.sh]: I have not implemented your request as I need more information about the use case to understand.
I was under the assumption that this code would execute once `when the container is being created`.

Is your comment about covering the case when the container is already living, that the user had manually removed the replacement/merge folders and then runs the script manually ?

File: sources/assets/exegol/load_supported_setups.sh
Comment (gbe) [reply sources/assets/exegol/load_supported_setups.sh]: Same as above.

File: sources/assets/exegol/load_supported_setups.sh
Comment (gbe) [reply sources/assets/exegol/load_supported_setups.sh:170]: I added some checks, including one I had missed when merging with Exegol's default file. I was not checking that it existed.
I think I covered your comment and more (hopefully).

**Comment by gbe** [comment]
Besides my comment, you can also notice that I moved some paths to local variables, as I found it error-prone to have duplicates.

File: sources/assets/exegol/load_supported_setups.sh
Comment (gbe) [reply sources/assets/exegol/load_supported_setups.sh]: Ok, I think I now understand a bit better the use case. This request should be implemented the way you want.

File: sources/assets/exegol/load_supported_setups.sh
Comment (gbe) [reply sources/assets/exegol/load_supported_setups.sh]: Done.

**Pull request merged**
